src/app/context/wallet/WalletProvider.tsx

```diff
@@ -0,0 +1,160 @@
+import type { networks } from "bitcoinjs-lib";
+import {
+  type PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+import { ConnectModal } from "@/app/components/Modals/ConnectModal";
+import { useError } from "@/app/context/Error/ErrorContext";
+import { ErrorState } from "@/app/types/errors";
+import {
+  getPublicKeyNoCoord,
+  isSupportedAddressType,
+  toNetwork,
+} from "@/utils/wallet";
+import { WalletError, WalletErrorType } from "@/utils/wallet/errors";
+import { type WalletProvider as IWalletProvider } from "@/utils/wallet/wallet_provider";
+
+interface WalletContextProps {
+  walletProvider?: IWalletProvider;
+  network?: networks.Network;
+  publicKeyNoCoord: string;
+  address: string;
+  connected: boolean;
+  disconnect: () => void;
+  open: () => void;
+}
+
+const WalletContext = createContext<WalletContextProps>({
+  walletProvider: undefined,
+  network: undefined,
+  connected: false,
+  publicKeyNoCoord: "",
+  address: "",
+  disconnect: () => () => {},
+  open: () => {},
+});
+
+export const WalletProvider = ({ children }: PropsWithChildren) => {
+  const [walletProvider, setWalletProvider] = useState<IWalletProvider>();
+  const [network, setNetwork] = useState<networks.Network>();
+  const [publicKeyNoCoord, setPublicKeyNoCoord] = useState("");
```

Comment:
nit: I noticed some inconsistencies in declaring the types across our repo, for example: https://github.com/babylonlabs-io/simple-staking/blob/5fcdde1aee4d96a41b90f30ff87270f2af4f25ab/src/app/components/Modals/ConnectModal.tsx#L32-L33 do we want to keep the types or not?

Reply:
We will soon replace this logic with Tomo, so I decided to not refactor this part @jeremy-babylonlabs